### DISCUSSION TITLE
Transaction failed on WalletConnect v2 with metamask connection

### DISCUSSION THREAD

**Discussion by Doberman2029** [WalletConnect/walletconnect-monorepo#3408] [Developer Support]
**Describe the bug**
When using wallet connect v2 together with a metamask connection, data that contains a numeric value does not change from decimal to hexadecimal when encoded inside. And because of this, an "ERC721: owner query for nonexistent token" error occurs.

**SDK Version (if relevant)**
 - Client: Next.js v13.4.8. ethers v 5.5.2, typechain v 8.1.0, wagmi 1.3.7

**To Reproduce**
1. Connecting a Metamask wallet via Wallet Connect v2.
2. Calling the contract method, which contains the uint256 value (in my case, this is an approve)
3. When signing in the wallet, during the execution of the transaction, an "ERC721: owner query for nonexistent token" error occurs

**Expected behavior**
The transaction is correctly encoded and completed successfully

**Desktop (please complete the following information):**
 - OS: Windows 10, macOs Ventura
 - Browser: Chrome, Safari

**Smartphone (please complete the following information):**
 - Device: iPhone13, iPhone14, Samsung Galaxy S23
 - OS: ios 16.5.1. Android 13
 - Browser: Chrome, Safari

**Additional context**
Method abi 
![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/20417138/e1a30f27-7a6e-4fae-8638-db8949c5932f)
Example of tx signed by connection through metamask extensions (success)
https://mumbai.polygonscan.com/tx/0x929bb13fee7a594adc2ea5774db0baab3efc8d0002f5b021e53bc0054ace5fb8 
And connection by MM + WalletConnectv2 (not success)
https://mumbai.polygonscan.com/tx/0x52bb3ce85b1400a21aaee7c798ff28ca5af77d31d15fb62b4cb7bc0a00070944
In both cases the same data is transmitted. Connection by trust wallet + wallet connect v2 works correctly

**Comment by glitch-txs** [comment]
hi, could you share a minimal reproducible example of this issue?

**Reply by Doberman2029** [reply]
Hi, @glitch-txs. It's metamask problem, [here issue](https://github.com/MetaMask/metamask-mobile/issues/6815). To reproduce you need to request an approval of any nft whose id is greater than 16. And you must be connected via metamask mobile browser
